var/spack/repos/builtin/packages/the-silver-searcher/package.py

```diff
@@ -21,3 +21,6 @@ class TheSilverSearcher(AutotoolsPackage):
     depends_on("xz")
     depends_on("zlib")
     depends_on("pkgconfig", type="build")
+
+    # https://github.com/spack/spack/issues/25664
+    patch("fno-common.patch", when="@2.2.0")
```

Comment:
Can you add a brief comment above the `patch` directive explaining:
1. What is the effect of the patch
2. Why it is needed
3. Where it was taken from

?